diff --git a/pathauto.module b/pathauto.module
index 5027dec..afb2b71 100644
--- a/pathauto.module
+++ b/pathauto.module
@@ -554,12 +554,13 @@ function pathauto_action_info() {
 function pathauto_entity_language($entity_type, $entity, $check_language_property = TRUE) {
   $langcode = NULL;
 
+  if ($check_language_property && !empty($entity->language)) {
+    return $entity->language;
+  }
+
   if (function_exists('entity_language')) {
     $langcode = entity_language($entity_type, $entity);
   }
-  elseif ($check_language_property && !empty($entity->language)) {
-    $langcode = $entity->language;
-  }
 
   return !empty($langcode) ? $langcode : LANGUAGE_NONE;
 }
